yt/data_objects/static_output.py

```diff
@@ -1243,29 +1244,83 @@ def set_code_units(self):
                 "unitary", float(DW.max() * DW.units.base_value), DW.units.dimensions
             )
 
+    @classmethod
+    def _validate_units_override_keys(cls, units_override):
+        valid_keys = set(cls.default_units.keys())
+        invalid_keys_found = set(units_override.keys()) - valid_keys
+        if invalid_keys_found:
+            raise ValueError(
+                f"units_override contains invalid keys: {invalid_keys_found}"
+            )
+
+    default_units = {
+        "length_unit": "cm",
+        "time_unit": "s",
+        "mass_unit": "g",
+        "velocity_unit": "cm/s",
+        "magnetic_unit": "gauss",
+        "temperature_unit": "K",
+    }
+
+    @classmethod
+    def _sanitize_units_override(cls, units_override):
+        """
+        Convert units_override values to valid input types for unyt.
+        Throw meaningful errors early if units_override is ill-formed.
+        """
+        uo = {}
+        if units_override is None:
+            return uo
+
+        cls._validate_units_override_keys(units_override)
+
+        for key in cls.default_units:
+            try:
+                val = units_override[key]
+            except KeyError:
+                continue
+            try:
+                uo[key] = YTQuantity(val)
+                continue
+            except RuntimeError:
+                pass
+            try:
+                uo[key] = YTQuantity(*val)
+                continue
+            except (RuntimeError, TypeError, UnitParseError):
+                pass
+            raise TypeError(
+                "units_override values should be 2-tuples (float, str), "
+                "YTQuantity objects or real numbers; "
+                "received {} with type {}.".format(val, type(val))
+            )
+        for key, q in uo.items():
+            if q.units.is_dimensionless:
```

Comment:
Does this work with magnetic fields, which seem to be the source of so many issues with the specifics of units?

Reply:
We should only end up here if the user defines a unit using pure number (I think). Could you elaborate what makes magnetic field special ?

Reply:
Forget I said anything; I think B-fields are handled just fine.  :)